src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -439,7 +441,35 @@ public int Seek(Vector<byte> byte0Vector, Vector<byte> byte1Vector, Vector<byte>
             }
         }
 
-        private static int FindFirstEqualByte(Vector<byte> byteEquals)
+        /// <summary>
+        /// Find 
+        /// </summary>
+        /// <param name="byteEquals"></param>
+        /// <returns>The first index of the result vector if found otherwise return int.MaxValue</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int FindFirstEqualByte(Vector<byte> byteEquals)
+        {
+            // Fallback to normal search when IsLittleEndian=false
+            if (!BitConverter.IsLittleEndian) return _FindFirstEqualByte(byteEquals);
+
+            var vector64 = Vector.AsVectorInt64(byteEquals);
+            for (var i = 0; i < Vector<long>.Count; i++)
+            {
+                var longValue = vector64[i];
+                if (longValue == 0) continue;
+
+                return (i << 3) +
+                    ((longValue & 0x00000000ffffffff) == 0
+                    ? (longValue & 0x0000ffff00000000) == 0
+                    ? (longValue & 0x00ff000000000000) == 0 ? 7 : 6
+                    : (longValue & 0x000000ff00000000) == 0 ? 5 : 4
+                    : (longValue & 0x000000000000ffff) == 0
+                    ? (longValue & 0x0000000000ff0000) == 0 ? 3 : 2
+                    : (longValue & 0x00000000000000ff) == 0 ? 1 : 0);
+            }
+            throw new InvalidOperationException();
+        }
+        public static int _FindFirstEqualByte(Vector<byte> byteEquals)
```

Comment:
Since this is being called from a test, maybe make this internal instead and add a comment that it's accessible for testing.